src/daemon/mod.rs

```diff
@@ -453,9 +463,16 @@ async fn fetch_snapshot_if_required(
     config: &mut Config,
```

Comment:
Let's radically simplify this function. There's no longer any point in downloading the snapshot here. Instead, just set the snapshot URL in `config.client.snapshot_path` and return.

Reply:
Fixed